### PR TITLE
Feature/update task.hcd 1002.hcd 1004 implement update task use case

### PR DESCRIPTION
**Pull request by DomeJaturawit**

### CODE REVIEW

File: domain/todo_entity_model.go [hunk 1]
```diff
@@ -35,10 +35,25 @@ type GetTodoEntity struct {
...
+	Description string    `json:"description" gorm:"column:description"`
+	Status      string    `json:"status" gorm:"column:status"`
+	UpdatedAt   time.Time `json:"updated_at" gorm:"column:updated_at"`
+}
+
+func NewUpdateTodoEntity(title string, description string, status string, updatedAt time.Time) UpdateTodoEntity {
+	return UpdateTodoEntity{Title: title, Description: description, Status: status, UpdatedAt: updatedAt}
+}
+
+type QueryUpdateTodoEntity struct {
```
Comment (oat-eknarin-hugeman) [inline domain/todo_entity_model.go]: Please rename to UpdateTodoQueryEntity

File: repository/model/repository_model.go [hunk 2]
```diff
@@ -6,14 +6,35 @@ import (
...
+type TbTodoRepositoryCreateModel struct {
 	ID          uuid.UUID `json:"id" gorm:"primary_key"`
 	Title       string    `json:"title"`
 	Description string    `json:"description"`
 	Status      string    `json:"status"`
 	CreatedAt   time.Time `json:"created_at"`
 }
 
-func (TbTodoRepositoryModel) TableName() string {
+type TbTodoRepositoryUpdateModel struct {
```
Comment (oat-eknarin-hugeman) [inline repository/model/repository_model.go:17]: Please define all update fields as pointer incase of we want to update some fields

File: usecase/update_todo_usecase.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		return nil, errorLib.WrapError(common.ErrBeginTodo.Error(), err)
+	}
+	result, err = n.repo.UpdateTodoRepository(ctx, dbTx, queryEntity, domain.UpdateTodoEntity{
+		Title:       entity.Title,
+		Description: entity.Description,
+		Status:      entity.Status,
+		UpdatedAt:   time.Now(),
+	})
+	if err != nil {
+		return nil, errorLib.WrapError(common.ErrUseCaseUpdateTodo.Error(), err)
```
Comment (oat-eknarin-hugeman) [inline usecase/update_todo_usecase.go:32]: If error you must rollback transaction

**Review by oat-eknarin-hugeman — COMMENTED** [review]
Please review my comments kub

File: usecase/update_todo_usecase.go (hunk 3 above)
Comment (DomeJaturawit) [reply usecase/update_todo_usecase.go:32]: i use rollback in line 33

Comment (oat-eknarin-hugeman) [reply usecase/update_todo_usecase.go:32]: In line 33 will handle from error repo.Commit
That's why you need to rollback in repo.UpdateToDoRepository too

**Review by oat-eknarin-hugeman — APPROVED** [review]

**Pull request merged**